crates/interledger-stream/src/error.rs

```diff
@@ -1,12 +1,35 @@
+use interledger_packet::{AddressError, ErrorCode, PacketTypeError as IlpPacketTypeError};
 /// Stream Errors
+use std::str::Utf8Error;
+
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
-    #[error("Error connecting: {0}")]
-    ConnectionError(String),
-    #[error("Error polling: {0}")]
-    PollError(String),
-    #[error("Error polling: {0}")]
-    SendMoneyError(String),
-    #[error("Error maximum time exceeded: {0}")]
-    TimeoutError(String),
+    #[error("Terminating payment since too many packets are rejected ({0} packets fulfilled, {1} packets rejected)")]
+    PaymentFailFast(u64, u64),
+    #[error("Packet was rejected with ErrorCode: {0}")]
+    SendMoneyPacket(ErrorCode),
+    #[error("Error maximum time exceeded since last fulfill")]
+    Timeout,
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum StreamPacketError {
+    #[error("Unable to decrypt message")]
+    FailedToDecrypt,
+    #[error("Stream version not supported")]
+    UnsupportedVersion,
+    #[error("Frames Error: Not enough successfully parsed frames")]
+    NotEnoughValidFrames,
+    #[error("Trailing bytes error: Inner")]
+    TrailingInnerBytes,
+    #[error("Roundtrip only: Error expected for roundtrip fuzzing")]
+    Roundtrip,
+    #[error("I/O Error: {0}")]
+    IoErr(#[from] std::io::Error),
+    #[error("Ilp PacketType Error: {0}")]
+    IplPacketType(#[from] IlpPacketTypeError),
```

Comment:
```suggestion
    IlpPacketType(#[from] IlpPacketTypeError),
```
Was about to suggest `InnerPacketType` but then I was wondering if it was the inner or outer again. I'm leaning on it being the outer. Might be a good one to comment what this is..

Reply:
Note sure what is inner / outer here. It is the same `PacketType` parsed in interledger-packet/packet.rs. I think these are explicitly ILP PacketType.

Reply:
Interestingly I am not finding this from the [rfc](https://github.com/interledger/rfcs/blob/master/0029-stream/0029-stream.md) either, but there were of course many references to Prepares, Fulfills and Rejects.

Reply:
Well, lets just leave it like this.